lib/secure_sockets/portable/freertos_plus_tcp/aws_secure_sockets.c

```diff
@@ -690,22 +680,27 @@ uint32_t ulApplicationGetNextSequenceNumber( uint32_t ulSourceAddress,
     static uint64_t ullKey = 0;
 
     /* Acquire a crypto session handle. */
-    xResult = prvSocketsGetCryptoSession( &xPkcs11Session,
+    xResult = prvSocketsGetCryptoSession( &xSessionLock,
+                                          &xPkcs11Session,
                                           &pxPkcs11FunctionList );
 
     if( 0 == xResult )
     {
+        xSemaphoreTake( xSessionLock, portMAX_DELAY );
+
         if( 0 == ullKey )
         {
             /* One-time initialization, per boot, of the random seed. */
             xResult = pxPkcs11FunctionList->C_GenerateRandom( xPkcs11Session,
```

Comment:
Do we want to continue to exclude the possibility of a "key" value being 0? 

I realize that this is a restriction of ulRand that we are stuck with atm for backward compatibility, and that the odds of this occurring are very low, but could we initialize ullKey at the same time as we initialize all the other stuff?

This would also require us to find a way to access it across the different functions which might be a pain.

Reply:
Fixed.

Reply:
Fixed.